visualization/app/codeCharta/util/scenarioHelper.ts

```diff
@@ -49,14 +49,17 @@ export class ScenarioHelper {
 		return scenarioItems
 	}
 
-	private static isScenarioAppliable(scenario: RecursivePartial<Scenario>, metricData: NodeMetricData[]) {
```

Comment:
```suggestion
    private static isScenarioAppliable(scenario: RecursivePartial<Scenario>, metricData: MetricData) {
        const expected = Object.keys(scenario).filter(x => x === "area" || x === "height" || x === "color" || x === "edge").length
        let existingNodeMetricData = 0
        let existingEdgeMetricData = 0

        for (const nodeMetricData of metricData.nodeMetricData) {
            if(nodeMetricData.name === scenario.area?.areaMetric || nodeMetricData.name === scenario.color?.colorMetric
                || nodeMetricData.name === scenario.height?.heightMetric) {
                existingNodeMetricData++
            }
        }

        for (const edgeMetricData of metricData.edgeMetricData) {
            if(edgeMetricData.name === scenario.edge?.edgeMetric) {
                existingEdgeMetricData++
            }
        }

        return existingNodeMetricData + existingEdgeMetricData === expected
    }
```

This is slightly more performant. This only traverse each metricData once instead of traversing the nodeMetricData three times.
Better apply this in the IDE to ensure formatting the code.

Reply:
I would actually use a different approach:

```js
const { area, color, height } = scenario

if (area || color || height) {
  const nodeMetricSet = new Set(metricData.nodeMetricData.map((data) => data.name)
  
  if (area && !nodeMetricSet.has(area.areaMetric)) {
    return false
  }
  ...
}
```

I'd keep the suggested edge metric detection as it's the fasted implementation (we do not always have to go through all of them).

Reply:
``  if(nodeMetricData.name === scenario.area?.areaMetric || nodeMetricData.name === scenario.color?.colorMetric
                || nodeMetricData.name === scenario.height?.heightMetric) {
                existingNodeMetricData++
            }``

I like the idea of iterating just once through the nodeMetricData, but we have to find another solution for the if statement. The problem is, that when areaMetric and colorMetricfor exampe have the same metric saved, it will just increment +1 instead of +2.
That way it will return false as the expected value is 3.